hit-testing-explainer.md

```diff
@@ -0,0 +1,275 @@
+
+## Real-world hit testing
+A key challenge with enabling real-world hit testing in WebXR is that computing real-world hit test results can be performance-impacting and dependant on secondary threads in many of the underlying implementations. However from a developer perspective, out-of-date asynchronous hit test results are often, though not always, less than useful. 
+
+WebXR addresses this challenge through the use of the `XRHitTestSource` type which acts somewhat like a subscription mechanism. The presence of an `XRHitTestSource` signals to the user agent that the developer intends to query `XRHitTestResult`s in subsequent `XRFrame`s. The user agent can then precompute `XRHitTestResult`s based on the `XRHitTestSource` properties such that each `XRFrame` will be bundled with all "subscribed" hit test results. When the last reference to the `XRHitTestSource` has been released, the user agent is free to stop computing `XRHitTestResult`s for future frames.
+
+### Requesting a hit test source
+To create an `XRHitTestSource` developers call the `XRSession.requestHitTestSource()` function. This function accepts an `XRHitTestOptionsInit` dictionary with the following values:
+* `space` is required and is the `XRSpace` to be tracked by the hit test source. As this `XRSpace` updates its location each frame, the `XRHitTestSource` will move with it. If the supplied `space` is an `XRReferenceSpace`, the `offsetRay` of the resulting `XRHitTestSource` will be updated each time the `XRReferenceSpace.originOffset` is updated, even mid-frame.
+* `offsetRay` is optional and, if provided, is the `XRRay` from which the hit test should be performed. The ray's coordinates are defined with `space` as the origin. If an `offsetRay` is not provided, hit testing will be performed using a ray with coincident with the `space` origin and pointing in the "forward" direction. For more information on constructing an `XRRay`, see the [Spatial Tracking explainer](spatial-tracking-explainer.md#rays).
+
+In this example, an `XRHitTestSource` is created slightly above the center of the `viewerSpace`. This is because the developer has chosen to leave space for buttons along the bottom of the hand-held AR device screen while still giving the perception of a centered cursor. For more information, see [Rendering cursors and highlights](#rendering-cursors-and-highlights) in the Input Explainer.
```

Comment:
> the developer has chosen to leave space for buttons along the bottom of the hand-held AR device screen

I think this is a bit confusingly phrased. Buttons sound like DOM elements, and "leaving space" could be interpreted as shrinking the camera view to make room for them, and both aren't currently supported in `immersive-ar` mode, right? Maybe something like _"the developer has chosen to draw user interface elements in the bottom part of the immersive view"_ would make it clearer that this would still be part of the fullscreen immersive mode.

Reply:
That's exactly what's being described ^_^  We know a solution is required for some form of DOM  overlay on hand-held AR, and that's being tracked in issue #400.  That said, until we have a concrete solution, I can understand the hesitation about overly specific language and will fix it.